File: 3.Offstand/#.Private/MS/off/app/src/main/java/com/example/off/MainActivity.java
package com.example.off;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import static com.example.off.SplashActivity.adapter;

public class MainActivity extends AppCompatActivity {

    ImageButton imagebtn;

    ImageView myImage;
    TextView myNameText;
    TextView winText;
    TextView loseText;

    LinearLayoutManager layoutManager;
    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        layoutManager = new LinearLayoutManager(this);

        recyclerView = (RecyclerView) findViewById(R.id.recyclerView);




        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(adapter);

        Log.d("@@", "@@" + adapter.getItem(0));
        Log.d("@@", "@@" + adapter.getItem(1));
        Log.d("@@", "@@" + adapter.getItem(2));
        Log.d("@@", "@@" + adapter.getItem(3));

        myImage = (ImageView) findViewById(R.id.myImage);
        myNameText = (TextView) findViewById(R.id.myName);
        winText = (TextView) findViewById(R.id.myWin);
        loseText = (TextView) findViewById(R.id.myLose);
        imagebtn = (ImageButton) findViewById(R.id.imagebtn);


        SharedPreferences sharedPreferences = getSharedPreferences("version", MODE_PRIVATE);

        myImage.setImageResource(sharedPreferences.getInt("myImage", 0));
        myNameText.setText(sharedPreferences.getString("myName", ""));
        winText.setText(Integer.toString(sharedPreferences.getInt("myWin", 0)));
        loseText.setText(Integer.toString(sharedPreferences.getInt("myLose", 0)));

        imagebtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(), SplashActivity.class));

            }
        });

        adapter.setOnItemClickListener(new ItemAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(ItemAdapter.ViewHolder holder, View view, int i) {
                Item item = adapter.getItem(i);

                Toast.makeText(getApplicationContext(), "UserId 선택됨 : " + item.getUserId(), Toast.LENGTH_SHORT).show();

            }
        });

    }
}

File: 4. Routine/#.Private/DDang/RoomDBTest/app/src/main/java/com/example/roomdbtest/MainActivity.kt
package com.example.roomdbtest

import android.content.Intent
import android.os.AsyncTask
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.lifecycle.ViewModel
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), PersonAdapter.OnItemClickListener<Person> {

    private var personDb: PersonDB? = null
    private var personList = listOf<Person>()
    lateinit var mAdapter: PersonAdapter

    public fun selectDB() {
        personList = personDb?.personDao()?.getAll()!!
        mAdapter.setItems(personList)
        mAdapter.notifyDataSetChanged()
    }

    public class PersonDeleteAsync(var personDb : PersonDB?) : AsyncTask<Person, Void, Void?>() {

        override fun onPreExecute() {
            super.onPreExecute()
        }

        override fun doInBackground(vararg person: Person?): Void? {
            personDb?.personDao()?.delete(person[0]?.id!!.toLong())
            return null
        }

        override fun onPostExecute(result: Void?) {
            super.onPostExecute(result)
        }
    }

    override fun onItemClick(person: Person, position: Int) {
        Toast.makeText(this, "${person.id} / ${person.name} / ${person.age} / ${person.sex}", Toast.LENGTH_SHORT).show()
        Log.d("test","onItemClick Thread ${Thread.currentThread().name}" )
//        personDb?.personDao()?.delete(person.id!!.toLong())
        var personDeleteAsync : PersonDeleteAsync
        personDeleteAsync = PersonDeleteAsync(personDb)
        personDeleteAsync.execute(person)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        personDb = PersonDB.getInstance(this)
        mAdapter = PersonAdapter(this, personList) {
//            Log.d("tag", "item Click")
        }

        val r = Runnable {
            try {
                mAdapter.setOnItemClickListener(this)
                selectDB()
                RoomRecyclerView.adapter = mAdapter
                RoomRecyclerView.layoutManager = LinearLayoutManager(this)
                RoomRecyclerView.setHasFixedSize(true)
            } catch (e: Exception) {
                Log.d("tag", "Error - $e")
            }
        }

        val thread = Thread(r)
        thread.start()

        addButton.setOnClickListener {
            val i = Intent(this, AddPerson::class.java)
            startActivity(i)
            finish()
        }
    }

    override fun onDestroy() {
        PersonDB.destroyInstance()
        personDb = null
        super.onDestroy()
    }
}
